compiler/hash-typecheck/src/exhaustiveness/lower.rs

```diff
@@ -135,123 +54,319 @@ impl<'tc> LowerPatOps<'tc> {
         Self { storage }
     }
 
-    /// Take a [primitives::Pat] and convert it into [Pat], essentially lowering
-    /// the pattern.
-    pub fn lower_pat(&self, id: PatId) -> Pat {
+    /// Convert a [Pat] into a [DeconstructedPat].
+    pub(crate) fn deconstruct_pat(&self, ty: TermId, pat_id: PatId) -> DeconstructedPat {
         let reader = self.reader();
+        let pat = reader.get_pat(pat_id);
+
+        let (ctor, fields) = match pat {
+            Pat::Wild
+            | Pat::Mod(_)
+            | Pat::Binding(_)
+            | Pat::Spread(_)
+            | Pat::Access(_)
```

Comment:
How is access a wildcard?

Reply:
It gets simplified to a wildcard anyway, there are no situations where it can be a non-equivalent of a binding

Reply:
what about `Option::Some` ? I guess it would get simplified to a `Pat::Const`?

Reply:
Yes it does get simplified to a `Pat::Const`

Reply:
But also this would be an invalid match... since it could never match the type of `Option::Some` with the subject

Reply:
I meant Option::None

Reply:
Yes, then the comment about access/const pats applies